### PR TITLE
Add example for Dynamax

### PR DESCRIPTION
**Pull request by theorashid**
Really basic example just to swap out the inference stage. Tried to keep it as short as possible.

### CODE REVIEW

**Comment by ColCarroll** [comment]
sorry, no great way set up for me to leave comments right on the PR, but this looks great. a few quick changes:

1. can you do the more standard seed splitting (and use new style keys!): 
```
seed = jax.random.key(0)
init_key, sample_key, test_init_key, inference_key = jax.random.split(seed, 4)
```

2. I'd probably leave out the advertisement to `.get_kwargs`, unless you feel strongly!

3. I think you need to call 
```
constrained_samples = from_unconstrained(samples, param_props)
```
to have your draws be back in the original space. I'm not sure why that doesn't happen in the dynamax notebook, and makes me nervous I'm wrong about that?

**Comment by theorashid** [comment]
You're spot on. They have `from_unconstrained()` within with hmc_step in their notebook

```python
    @jit
    def hmc_step(hmc_state, step_key):
        next_hmc_state, _ = hmc_kernel(step_key, hmc_state)
        params = from_unconstrained(hmc_state.position, props)
        return next_hmc_state, params
```

Pushing the other changes now
